server/socketserver.go

```diff
@@ -57,18 +56,23 @@ func (s *socketServer) execute(handler Handler, jsonBytes []byte) (reply []byte)
 		return b.Bytes()
 	}
 
-	reply, handlingError := handleRequest(s.repo, handler, jsonBytes, sourceSocketServer)
-	if handlingError != nil {
-		Log.Error("socketServer.execute failed", zap.Error(handlingError))
+	var buf bytes.Buffer
+	buf.Grow(len(jsonBytes))
+	if err := handleRequest(s.repo, handler, bytes.NewReader(jsonBytes), &buf, sourceSocketServer); err != nil {
+		Log.Error("socketServer.execute failed", zap.Error(err))
 	}
-	return reply
+	return buf.Bytes()
 }
 
-func (s *socketServer) writeResponse(conn net.Conn, reply []byte) {
-	headerBytes := []byte(strconv.Itoa(len(reply)))
+func (s *socketServer) writeResponse(w io.Writer, reply []byte) {
+	headerBytes := make([]byte, 0, len(reply)+12)
```

Comment:
Please add a comment that explains why the maximum header size is len(reply)+12

Reply:
done